cpp/src/io/csv/writer_impl.cu

```diff
@@ -0,0 +1,545 @@
+  }
+
+  // strings:
+  //
+  template <typename column_type>
+  std::enable_if_t<std::is_same<column_type, cudf::string_view>::value, std::unique_ptr<column>>
+  operator()(column_view const& column_v) const
+  {
+    using namespace cudf::strings::detail;
+
+    // handle special characters: {delimiter, '\n', "} in row:
+    //
+    // algorithm outline:
+    //
+    // target = "\"";
+    // repl = ""\"\";
+    //
+    // str_column_ref = {};
+    // for each str_row: column_v {
+    //    if ((not null str_row) &&
+    //        (str_row.find("\n") || str_row.find("\"") || str_row.find(delimiter) ))
+    //        str_column_modified = modify(str_row);
+    // where modify() = duplicate the double quotes, if any; add 2bl quotes prefix/suffix;
+    //}
+    //
+    std::string delimiter{options_.inter_column_delimiter()};
+    predicate_special_chars pred{delimiter, stream_};
+
+    return modify_strings<probe_special_chars, modify_special_chars>(column_v, mr_, stream_, pred);
+
+    // null replacement could be done here, but probably more efficient
+    // to defer it until concatenate() call:
+    //
+    // return cudf::strings::replace_nulls(column,
+    //                                     options_.na_rep() ,
+    //                                     mr_);
+  }
+
+  // ints:
+  //
+  template <typename column_type>
+  std::enable_if_t<std::is_integral<column_type>::value && !std::is_same<column_type, bool>::value,
+                   std::unique_ptr<column>>
+  operator()(column_view const& column) const
+  {
+    auto conv_col_ptr = cudf::strings::from_integers(column, mr_);
+
+    return conv_col_ptr;
+
+    // null replacement could be done here, but probably more efficient
+    // to defer it until concatenate() call:
+    //
+    // strings_column_view strings_converted{std::move(*conv_col_ptr)};
+    // return cudf::strings::replace_nulls(strings_converted,
+    //                                     options_.na_rep() ,
+    //                                     mr_);
+  }
+
+  // floats:
+  //
+  template <typename column_type>
+  std::enable_if_t<std::is_floating_point<column_type>::value, std::unique_ptr<column>> operator()(
+    column_view const& column) const
+  {
+    auto conv_col_ptr = cudf::strings::from_floats(column, mr_);
+
+    return conv_col_ptr;
+
+    // null replacement could be done here, but probably more efficient
+    // to defer it until concatenate() call:
+    //
+    // strings_column_view strings_converted{std::move(*conv_col_ptr)};
+    // return cudf::strings::replace_nulls(strings_converted,
+    //                                     options_.na_rep() ,
+    //                                     mr_);
+  }
+
+  // timestamps:
+  //
+  template <typename column_type>
+  std::enable_if_t<cudf::is_timestamp<column_type>(), std::unique_ptr<column>> operator()(
+    column_view const& column) const
+  {
+    std::string format{"%Y-%m-%dT%H:%M:%SZ"};  // same as default for `from_timestamp`
```

Comment:
Since the delimiter can be passed by the caller, there could be a case where the delimiter is "-" or ":" and therefore require this column to have quotes around it because the `format` string contains the delimiter. One option is to check the delimiter and substitute something else in `format` before calling `from_timestamp()` or maybe just remove the dividing characters altogether `"%Y%m%dT%H%M%SZ"`
If this is out of scope then perhaps create an issue to fix it later.

Reply:
OK, fixed.